compiler/infra/ILWalk.hpp

```diff
@@ -60,15 +59,17 @@ class TreeTopIteratorImpl
 
    // For logging
    //
+   TR::Compilation  *_comp;
    const char       *_name; // NULL means we don't want logging
-   TR::Optimization *_opt;
+
+   TR::Compilation * comp() { return _comp; }
 
    void logCurrentLocation();
 
    void stepForward();
    void stepBackward();
 
-   TreeTopIteratorImpl(TR::TreeTop *start, TR::Optimization *opt=NULL, const char *name=NULL);
+   TreeTopIteratorImpl(TR::TreeTop *start, TR::Compilation *comp=NULL, const char *name=NULL);
```

Comment:
In the original implementation, it was the `TR::Optimization` that was optional. I just changed the type of the parameter.

I don't think it makes sense for these APIs to be used outside of a compilation, since they operate exclusive on the compiler's IL, which is only (currently) available during compilation.

Personally, I would prefer if the parameter was mandatory, but I think making this change is outside the scope of this already large pull request.